src/toil_scripts/lib/test/test_programs.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+
+def test_which():
+    from toil_scripts.lib.programs import which
+    assert which('python')
+
+
+def test_docker_call(tmpdir):
+    from toil_scripts.lib.programs import docker_call
+    work_dir = str(tmpdir)
+    parameter = ['--help']
+    tool = 'quay.io/ucsc_cgl/samtools'
+    docker_call(work_dir=work_dir, parameters=parameter, tool=tool)
+    # Test outfile
+    fpath = os.path.join(work_dir, 'test')
+    with open(fpath, 'w') as fout:
+        docker_call(work_dir=work_dir, parameters=parameter, tool=tool, outfile=fout)
+    assert os.path.getsize(fpath) > 0
+    # Test env
+    cmd = docker_call(work_dir=work_dir, parameters=parameter, tool=tool, env=dict(JAVA_OPTS='-Xmx15G'))
+    assert '-e JAVA_OPTS=-Xmx15G' in ' '.join(cmd)
```

Comment:
While that command works from my terminal, I cannot get "$foo" to be expressed when run from subprocess — I've tried putting single quotes around it, running it with shell=True, and using curly braces.  If i run it with shell=True, stdout returns an empty string.  If i run it with shell=False, it returns $foo instead of bar.   

The only way i could get it to work was by directly passing the env variable through subprocess (not docker), and executing with shell=True. 

```
new_env = os.environ.copy()
new_env['foo'] = bar
p = subprocess.Popen("docker run --rm --log-driver=none ubuntu echo -n $foo", shell=True, stdout=subprocess.PIPE, env=new_env)
p.communicate()
>>> ('bar', None)
```

Any suggestions?